### PR TITLE
[Phase 1] LPS-178022 Remove Optional and Stream usages for module: portal-search-web Part 2 Final

### PR DESCRIPTION
**Pull request by julschong**
@dantewang Hi Dante, This PR avoids changing direct -web-api Optional usages.
After this is done, there is 1 more final -web-api change and we are done!

### CODE REVIEW

**Comment by julschong** [comment]
ci:test:search

File: modules/apps/portal-search/portal-search-web/src/main/java/com/liferay/portal/search/web/internal/custom/facet/portlet/CustomFacetPortletPreferencesImpl.java [hunk 1]
```diff
@@ -35,29 +35,29 @@ public CustomFacetPortletPreferencesImpl(
...
 		return _portletPreferencesHelper.getString(
 			CustomFacetPortletPreferences.PREFERENCE_KEY_AGGREGATION_FIELD);
 	}
 
 	@Override
-	public String getAggregationFieldString() {
-		Optional<String> optional = getAggregationFieldOptional();
-
-		return optional.orElse(StringPool.BLANK);
+	public String getAggregationField(String defaultValue) {
```
Comment (dantewang) [inline modules/apps/portal-search/portal-search-web/src/main/java/com/liferay/portal/search/web/internal/custom/facet/portlet/CustomFacetPortletPreferencesImpl.java:44]: @julschong For all the versions of the methods with `defaultValue` parameter in these `*PortletPreferencesImpl`, do you always use it with `StringPool.BLANK` as the default value?

If so, just wrap the non-default version with `GetterUtil.getString()`.

For example,
```
this.getAggregationField(StringPool.BLANK) -> GetterUtil.getString(this.getAggregationField())
```

**Review by dantewang — CHANGES_REQUESTED** [review]

**Comment by julschong** [comment]
ci:test:sf

**Comment by julschong** [comment]
ci:test:relevant

**Comment by julschong** [comment]
ci:test:search

**Comment by julschong** [comment]
pushed to change OrBlank method to use GetterUtil.getString instead

Fixed regression by using !Validator.isBlank instead of != null in SearchStringUtil

**Comment by Kvale1733** [comment]
@julschong Regressions found

**1) User search does not redirect to my profile page**

Steps to reproduce:
1. Go to default home page
2. User search widget to search for default admin user (Most likely Test Test)
3. Click on default admin user in search results

Expected Results: User is redirected to the My Profile page 
Actual Results: User is taken to default site with blank page

**2) NPE when adding search widget after adding Low Level Search Options portlet**

Steps to reproduce:
1. Add a test site with a test widget page
2. Add a Low Level Search Options widget to the page
3. Add a Search Results widget to the page

Expected Results: Can add the second search widget without issues
Actual Results: The search results widget shows "Portlet is temporarily unavailable" and the following appears in the console
```
2023-04-06 16:49:29.287 ERROR [http-nio-8080-exec-6][render_portlet_jsp:131] null
java.lang.NullPointerException: null
	at com.liferay.portal.search.web.internal.util.SearchStringUtil.splitAndUnquote(SearchStringUtil.java:39) ~[?:?]
	at com.liferay.portal.search.web.internal.low.level.search.options.portlet.shared.search.LowLevelSearchOptionsPortletSharedSearchContributor.contribute(LowLevelSearchOptionsPortletSharedSearchContributor.java:65) ~[?:?]
	at com.liferay.portal.search.web.internal.portlet.shared.search.PortletSharedSearchRequestImpl.lambda$_getSearchSettingsContributor$3(PortletSharedSearchRequestImpl.java:248) ~[?:?]
	at com.liferay.portal.search.web.internal.search.request.SearchRequestImpl.lambda$_buildSettings$1(SearchRequestImpl.java:121) ~[?:?]
	at java.lang.Iterable.forEach(Iterable.java:75) ~[?:1.8.0_362]
	at com.liferay.portal.search.web.internal.search.request.SearchRequestImpl._buildSettings(SearchRequestImpl.java:120) ~[?:?]
	at com.liferay.portal.search.web.internal.search.request.SearchRequestImpl.search(SearchRequestImpl.java:65) ~[?:?]
	at com.liferay.portal.search.web.internal.portlet.shared.search.PortletSharedSearchRequestImpl._search(PortletSharedSearchRequestImpl.java:295) ~[?:?]
	at com.liferay.portal.search.web.internal.portlet.shared.search.PortletSharedSearchRequestImpl.lambda$search$0(PortletSharedSearchRequestImpl.java:77) ~[?:?]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) ~[?:1.8.0_362]
	at com.liferay.portal.search.web.internal.portlet.shared.task.PortletSharedTaskExecutorImpl.executeOnlyOnce(PortletSharedTaskExecutorImpl.java:63) ~[?:?]
```
[Log](https://testray.liferay.com/reports/production/logs/2023-04/test-1-26/test-portal-acceptance-pullrequest(master)/8590/functional-tomcat90-mysql57-jdk8/2/10/jenkins-console.txt.gz?authuser=0)

3) Several checkboxes that were checked by default are no longer checked. For example

Steps to reproduce: 
1. Create a test widget page
2. Add a Search Results Widget
3. Click on the top right ellipses in the widget and then click Configuration

Expected Results: "Enable Highlighting" should be checked by default
Actual Results: "Enable Highlighting" is not checked by default

4) Some search configurations are not maintained after upgrade. All upgrades tests for search portlets fail at this checkbox assertion.

Steps to reporudce
1. Import the following database dump: https://github.com/liferay/liferay-qa-portal-legacy-ee/blob/master/7.3.0/data-archive/data-archive-search-portlets-mysql.zip
2. Add "upgrade.database.auto.run=true" and startup the portal so the database auto upgrades
3. Go to the "Site Page" page of site "Site Name"
4. Click on the ellipses for the Search Options widget and click Configuration

Expected Results: "Allow empty searches" should be checked as it was checked in the legacy database
Actual Results "Allow empty searches" is not checked.

**Comment by julschong** [comment]
see https://github.com/liferay-core-infra/liferay-portal/pull/4157

**Pull request closed without merging**